homeassistant/components/openweathermap/weather_update_coordinator.py

```diff
@@ -83,181 +78,122 @@ async def _async_update_data(self):
         data = {}
         async with asyncio.timeout(20):
             try:
-                weather_response = await self._get_owm_weather()
-                data = self._convert_weather_response(weather_response)
-            except (APIRequestError, UnauthorizedError) as error:
+                weather_report = await self._owm_client.get_weather(
+                    self._latitude, self._longitude
+                )
+                data = self._convert_weather_response(weather_report)
```

Comment:
Please move code out of the `try` block that cannot raise the exception being caught

Reply:
done, I hope I understood well what you mean